### ISSUE TITLE
Possible false positives in RedundantNilCheckRule

### ISSUE THREAD

**Issue by maplesteve** [oclint/oclint#141]
The `RedundantNilCheckRule` complains for the following code

```
if (object && [object message])
```

with a `redundant nil check` violation which I think is not correct, if `message` returns a `BOOL`. Sending messages to nil objects returns (always) `nil` which equals a boolean `FALSE`. If the nil-check is left out, the result of the expression can be incorrect.
I created a test class to show the behaviour. It fails in test cases 5 and 7 which shouldn't be the case. Please have a look at the test class here: https://gist.github.com/maplesteve/7415806

What do you think?
